spk/tvheadend/src/service-setup.sh

```diff
@@ -72,13 +66,13 @@ service_preupgrade ()
         done
         # Move recording directories from package root into var
         UPGRADE_CFG_DIR="${SYNOPKG_PKGDEST}/var/dvr/config"
-        echo "Move any recording directories from package root directory into var..." >> ${INST_LOG}
+        echo "Move any recording directories from package root directory into var..."
         for file in ${UPGRADE_CFG_DIR}/*
         do
-            DVR_DIR=`grep -e 'storage\":' ${file} | awk -F'"' '{print $4}'`
-            REAL_DIR=`realpath "${DVR_DIR}"`
-            TRUNC_DIR=`echo "${REAL_DIR}" | cut -c9-28`
-            CHECK_VAR=`echo "${REAL_DIR}" | cut -c29-32`
+            DVR_DIR=$(grep -e 'storage\":' ${file} | awk -F'"' '{print $4}')
+            REAL_DIR=$(realpath "${DVR_DIR}")
+            TRUNC_DIR=$(echo "${REAL_DIR}" | cut -c9-28)
+            CHECK_VAR=$(echo "${REAL_DIR}" | cut -c29-32)
             # Replace any remaining target links in log files first
             sed -i -e "s,/var/packages/tvheadend/target,/usr/local/tvheadend,g" "${file}"
```

Comment:
And here it would be better off using `${SYNOPKG_PKGDEST}` par of sed.

Reply:
This `sed` command will work in DSM7 but the path won't be available. Why is it needed anyway? Why can't the app do its thing in `/var/packages/tvheadend`?

Reply:
Honestly I have no clue.
I guess this `sed` could just be removed entirely.

Reply:
I tried to find a sample in `SYNOPKG_PKGVAR}/dvr/log` but I don't see any. I guess I need to have a receiver and start a recording.

Reply:
But I do :)

Got the following on my DSM6:
```
root@th0ma7-nas:/var/packages/tvheadend/target/var/dvr# tree -d
.
├── autorec
├── config
└── log

3 directories
```
Honestly I've been wishing to rework all this for a year but never took the time to do so.  The backup/recovery process does not work, period.

1. For instance the `/var/packages/tvheadend/target/var/config` references to `/usr/local/tvheadend` should be `sed`ed to `/var/package/tvheadend` (clearly the "`binary matches`" isn't good in the grep below):
```
# grep -R "/usr/local/tvheadend" *
Binary file bin/tvheadend matches
var/channel/config/65ab00d40fe4addffdf4babbc3b390d4:	"icon": "file://usr/local/tvheadend/.xmltv/",
var/channel/config/f87d3e9167eceb6d6f13e9ab7739f0cb:	"icon": "file://usr/local/tvheadend/.xmltv/",
var/channel/config/0cdf1b095014ab7990f9c8569a69270b:	"icon": "file://usr/local/tvheadend/.xmltv/",
var/channel/config/1fbe2fe151b5c6d9f555a69ce61af02f:	"icon": "file://usr/local/tvheadend/.xmltv/",
var/channel/config/3b1392ac0539b5b4a400b0d81d4e34eb:	"icon": "file://usr/local/tvheadend/.xmltv/",
var/config:	"chiconpath": "file://usr/local/tvheadend/.xmltv/",
var/config:	"muxconfpath": "/usr/local/tvheadend/share/tvheadend/data/dvb-scan/",
```
2. This entire directory (e.g. `/var/packages/tvheadend/target/var`) needs to be backup/restored along with creating a recovery `something-config-backup-date.tar.bz` in order to recover as needed.
My personal backup script: https://raw.githubusercontent.com/th0ma7/synology/master/tvheadend-backup.sh
3. There is a caching directory `/var/packages/tvheadend/target/cache` that needs to be recovered as well.  It would benefit from being located elsewhere under `/volume1/...` from a user defined param at installation time.
```
# du -hs cache var
24M	cache
14M	var
```

4. And also having a few scripts & configs that needs to be added manually that I should add to the default package...

BTW, just got my `tvheadend` and `ffmpeg` packages ready for DSM7... in theory ready for testing :)

Reply:
And the good: both packages install.
And the bad: Got a `503 Forbidden` after the password prompt when accessing the `http://<host>:9981/extjs.html` so guessing the password entry capture didn't worked as expected...

Reply:
@th0ma7 `503 Forbidden` is fixed with 72be92802f353cdf43a8108669dedf983c84d8bb

Reply:
I've just rebased to get the changes from master. If you want to try again.

Reply:
tried again but still `503 Forbidden`.
Also, I found what's really missing: setting the default storage option for recording.
This would allow removing almost the entire script and only focus on backuping the configs.
But, that could come into a later PR that I could work on once this one is in.

Reply:
Did you do a clean install/uninstall? Since the rebase I can always login with `admin:admin`

Reply:
I ensured to do a `git pull` and rebuilt the `tvheadend` package for `kvmx64`, fully uninstall/install (there is a new option now to completely wipe the app), kept passwd to default (`admin`) and still no luck.
I'll try to investigate this further later on.

Reply:
I found what the issue is... It relates to the changes of `${SYNOPKG_PKGDEST}` vs `${SYNOPKG_PKGVAR}`...

Directories `accesscontrol` and `passwd` are now located under `/var/packages/tvheadend/target/var -> /volume1/@appstore/tvheadend/var`) where the `password/a927e30a755504f9784f23a4efac5109` looks like:
```
{
	"enabled": true,
	"username": "admin",
	"password2": "@password@",
	"wizard": true
}
```
Moving both directories (`accesscontrol` and `passwd`) to `/var/packages/tvheadend/var -> /volume1/@appdata/tvheadend`) then manually fixing the password solved the isse:
```
mv /volume1/@appstore/tvheadend/var/* /volume1/@appdata/tvheadend/.
wizard_password=$(echo -n "TVHeadend-Hide-${wizard_password:=admin}" | openssl enc -a)
sed -i -e "s/@password@/${wizard_password}/g" /var/packages/tvheadend/var/a927e30a755504f9784f23a4efac5109
```

Reply:
@publicarray reminder on this.
I've looked at compile options but can't see something obvious that allows moving around theses two directories...
I'll try to have another look at it later and see if I can come-up with a fix somehow.
```Installation Paths
  --prefix=DIR$                  Installation root [/usr/local]
  --bindir=DIR                   Install binaries in DIR [${prefix}/bin]
  --libdir=DIR                   Install libraries in DIR [${prefix}/lib]
  --mandir=DIR                   Install man pages in DIR [${datadir}/man]
  --datadir=DIR                  Install data files in DIR [${prefix}/share]

```

Reply:
@publicarray and @hgy59 I believe I have found where the issue is.
And may have found a second one actually... or food for thought.

**First issue:**
It relates to `$(STAGING_DIR)` where the entire content is being `tgz` and ends-up being installed over to `/var/packages/tvheadend/target/` (_appstore_).  Although it happens that `work-*/staging/var` containing both `accesscontrol` and `passwd` that should follow the $(SYNOPKG_PKGVAR) and end-up into `/var/packages/tvheadend/var` (_appdata_) where all the automatically generated files also end-up going.

An idea to circumvent this:
1. From the `Makefile`, install files under `staging/appdata` using a new variable such as `$(STAGING_SPKVAR)`.
2. installer script go over all files & directories in `staging/appdata`.  If `install` just copy to `/var/packages/tvheadend/var` else perhaps something as trivial as a `rsync` no-overwrite over the directory (which could be used for all cases actually).  Depending of DSM the destination folder is set from `SYNOPKG_PKGVAR` so it should always work.


**Second issue: (not sure about this one)**
I wonder if it would make sense to extend solution proposed for the first issue to add variables for `$(STAGING_SPKHOME)`, `$(STAGING_SPKETC)` and `$(STAGING_SPKTEMP)` and use `rsync` to copy files over the end-state location?

Reply:
Or a simple work-around is to move the accesscontrol & passwd directory creation and file copying at the service-setup.sh level.

Reply:
@hgy59 and @publicarray Got a working solution.
Will provide a separate PR to address this later today.